kernel/fatal.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2019 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_internal.h>
+#include <sys/printk.h>
+#include <arch/cpu.h>
+#include <logging/log_ctrl.h>
+#include <logging/log.h>
+#include <fatal.h>
+
+/* LCOV_EXCL_START */
+FUNC_NORETURN __weak void z_arch_system_halt(unsigned int reason)
+{
+	ARG_UNUSED(reason);
+
+	/* TODO: What's the best way to totally halt the system if SMP
+	 * is enabled?
+	 */
+
+	(void)z_arch_irq_lock();
+	for (;;) {
+		k_cpu_idle();
+	}
+}
+/* LCOV_EXCL_STOP */
+
+/* LCOV_EXCL_START */
+__weak void k_sys_fatal_error_handler(unsigned int reason,
+				      const z_arch_esf_t *esf)
+{
+	ARG_UNUSED(esf);
+
```

Comment:
if this is the default place for unrecoverable error, i would put `LOG_PANIC` here

Reply:
Done